packages/mongodb-chatbot-server/src/routes/conversations/addMessageToConversation.ts

```diff
@@ -31,13 +32,29 @@ import {
   AddCustomDataFunc,
   ConversationsRouterLocals,
 } from "./conversationsRouter";
-import { GenerateUserPromptFunc } from "../../processors/GenerateUserPromptFunc";
-import { FilterPreviousMessages } from "../../processors/FilterPreviousMessages";
-import { filterOnlySystemPrompt } from "../../processors/filterOnlySystemPrompt";
-import { generateResponse, GenerateResponseParams } from "../generateResponse";
 import { wrapTraced } from "mongodb-rag-core/braintrust";
 import { UpdateTraceFunc, updateTraceIfExists } from "./UpdateTraceFunc";
 
+export type ClientContext = Record<string, unknown>;
+
+export interface GenerateResponseParams {
+  shouldStream: boolean;
+  latestMessageText: string;
+  clientContext?: ClientContext;
+  customData?: ConversationCustomData;
+  dataStreamer?: DataStreamer;
+  reqId: string;
+  conversation: Conversation;
+  request?: ExpressRequest;
+}
+
+export interface GenerateResponseReturnValue {
+  messages: SomeMessage[];
+}
+
+export type GenerateResponse = (
+  params: GenerateResponseParams
+) => Promise<GenerateResponseReturnValue>;
```

Comment:
Should this all be in a separate `generateResponse.ts` file? Seems like a distinct (if not separate) concern from the route that calls it.

Reply:
yeah reasonable, will do